scripts/qa_crowbarsetup.sh

```diff
@@ -3889,7 +3889,7 @@ function oncontroller_testsetup()
     wait_for 200 5 "openstack image show $imageid | grep status.*active" \
         "$image_name image to reach active state"
 
-    if [[ $want_ldap ]] ; then
+    if [[ $want_ldap = 1]] ; then
```

Comment:
I think we do `[[ $want_something ]]` in other places as well. I agree with you though that this way of writing the tests is error prone and should be avoided. Your expression is a lot more clear.